subsys/usb/device_next/usbd_ch9.c

```diff
@@ -520,41 +520,85 @@ static int sreq_get_desc_cfg(struct usbd_contex *const uds_ctx,
 	return 0;
 }
 
-static int sreq_get_desc(struct usbd_contex *const uds_ctx,
-			 struct net_buf *const buf,
-			 const uint8_t type, const uint8_t idx)
+/* Copy and convert ASCII-7 string descriptor to UTF16-LE */
+static void string_ascii7_to_utf16le(struct usbd_desc_node *const dn,
+				     struct net_buf *const buf, const uint16_t wLength)
+{
+	struct usb_string_descriptor *desc = dn->desc;
+	uint8_t *ascii7_str = (uint8_t *)&desc->bString;
+	size_t len;
+
+	LOG_DBG("wLength %u, bLength %u, tailroom %u",
+		wLength, desc->bLength, net_buf_tailroom(buf));
+
+	len = MIN(net_buf_tailroom(buf), MIN(desc->bLength,  wLength));
+
+	/* Add bLength and bDescriptorType */
+	net_buf_add_mem(buf, desc, MIN(len, 2U));
+	len -= MIN(len, 2U);
+
+	for (size_t i = 0; i < len; i++) {
+		__ASSERT(ascii7_str[i] > 0x1F && ascii7_str[i] < 0x7F,
+			 "Only printable ascii-7 characters are allowed in USB "
+			 "string descriptors");
```

Comment:
While this is the exact same text that was used earlier, it is somewhat confusing. The message here is not quite clear on the fact that the limitation comes from Zephyr USB stack and not USB Specification.